webknossos/tests/test_cli.py

```diff
@@ -0,0 +1,491 @@
+    in_path = TESTDATA_DIR / "tiff"
+
+    check_call(
+        "webknossos",
+        "convert",
+        "--jobs",
+        2,
+        "--voxel-size",
+        "11.24,11.24,25",
+        "--data-format",
+        str(data_format),
+        in_path,
+        out_path,
+    )
+
+    assert (out_path / "tiff").exists()
+    assert (out_path / "tiff" / "1").exists()
+
+
+@pytest.mark.skipif(
+    sys.platform != "linux",
+    reason="Only run this test on Linux, because it requires a running `minio` docker container.",
+)
+def test_tiff_cubing_zarr_s3() -> None:
+    """Tests zarr support when performing tiff cubing."""
+
+    out_path = REMOTE_TESTOUTPUT_DIR / "tiff_cubing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = MINIO_ROOT_PASSWORD
+    os.environ["AWS_ACCESS_KEY_ID"] = MINIO_ROOT_USER
+    os.environ["S3_ENDPOINT_URL"] = f"http://localhost:{MINIO_PORT}"
+
+    _tiff_cubing(out_path, DataFormat.Zarr)
+
+    assert (out_path / "tiff" / "1" / ".zarray").exists()
+    assert (out_path / PROPERTIES_FILE_NAME).exists()
+
+    with (out_path / PROPERTIES_FILE_NAME).open("r") as file, (
+        TESTDATA_DIR / "tiff" / "datasource-properties.zarr-fixture.json"
+    ).open("r") as fixture:
+        json_a = json.load(file)
+        json_fixture = json.load(fixture)
+        del json_a["id"]
+        del json_fixture["id"]
+        assert json_a == json_fixture
+
+
+def test_main() -> None:
+    """Tests the functionality of the webknossos command."""
+
+    result_without_args = runner.invoke(app, [])
+    assert result_without_args.exit_code == 0
+
+
+def test_check_equality() -> None:
+    """Tests the functionality of check_equality subcommand."""
+
+    result_without_args = runner.invoke(app, ["check-equality"])
+    assert result_without_args.exit_code == 2
+
+    result = runner.invoke(
+        app,
+        [
+            "check-equality",
+            str(TESTDATA_DIR / "simple_wkw_dataset"),
+            str(TESTDATA_DIR / "simple_wkw_dataset"),
+        ],
+    )
+    assert result.exit_code == 0
+    assert (
+        f"The datasets {str(TESTDATA_DIR / 'simple_wkw_dataset')} and "
+        f"{str(TESTDATA_DIR / 'simple_wkw_dataset')} are equal"
+        in result.stdout.replace("\n", "")
+    )
+
+
+def test_check_not_equal() -> None:
+    """Tests that the check_equality subcommand detects differing datasets."""
+
+    with tmp_cwd():
+        tmp_path = Path("testdata") / "simple_wkw_dataset"
+
+        copytree(TESTDATA_DIR / "simple_wkw_dataset", tmp_path)
+        dataset = Dataset.open(tmp_path)
+        finest_mag = dataset.get_layer("color").get_finest_mag().read()
+        finest_mag[1, 5, 7, 4] += 1
+        dataset.get_layer("color").get_finest_mag().write(finest_mag)
+
+        result = runner.invoke(
+            app,
+            [
+                "check-equality",
+                str(TESTDATA_DIR / "simple_wkw_dataset"),
+                str(tmp_path),
+            ],
+        )
+        assert result.exit_code == 0
```

Comment:
I think for differing datasets, the exit code should not be 0. Otherwise it is hard to use this command in bash scripts.